src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -169,6 +169,23 @@ protected void AssertSelf ()
 			throw new ObjectDisposedException (nameof (AndroidClientHandler));
 		}
 
+		string EncodeUrl (Uri url)
+		{
+			if (url == null)
+				return String.Empty;
+
+			if (String.IsNullOrEmpty (url.Query))
+				return Uri.EscapeUriString (url.ToString ());
+
+			// UriBuilder takes care of encoding everything properly
+			var bldr = new UriBuilder (url);
+			if (bldr.Port == 80)
```

Comment:
If you're going to check for Port 80 for "cleaner" urls, shouldn't you also check for port 443 (https)?

    csharp> var b = new UriBuilder ("https://example.org");
    csharp> b.Port;
    443
    csharp> b.ToString ();
    "https://example.org:443/"